File: app/src/main/java/com/example/bmicalculator/HowManyCalories.java
package com.example.bmicalculator;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.View;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

public class HowManyCalories extends AppCompatActivity {

    private Button mBtGoBack;
    private RadioGroup radioSexGroup;
    private RadioButton radioSexButton;
    private Button btnDisplay;
    private String sex;
    private TextView weight, height, age, youCanEat;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_how_many_calories);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mBtGoBack = findViewById(R.id.goBackBtn2);
        weight = findViewById(R.id.weightHarrisBenedict);
        height= findViewById(R.id.heightHarrisBenedict);
        age = findViewById(R.id.ageHarrisBenedict);
        youCanEat = findViewById(R.id.youCanEatText);
        radioSexGroup = findViewById(R.id.radioSexGroup);
        btnDisplay= findViewById(R.id.showWhat);


        addListenerOnButtons();
    }

    public void addListenerOnButtons(){

        btnDisplay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ShowMinimalCalories();
            }
        });

        mBtGoBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    private void  ShowMinimalCalories(){
        String heightString = height.getText().toString();
        String weightString = weight.getText().toString();
        String ageString = age.getText().toString();

        int selectedId= radioSexGroup.getCheckedRadioButtonId();
        radioSexButton = findViewById(selectedId);
        sex = radioSexButton.getText().toString();

        if(!heightString.isEmpty() && !weightString.isEmpty() && !ageString.isEmpty()){
            float heightValue= Float.parseFloat(heightString);
            float weightValue = Float.parseFloat(weightString);
            int ageValue= Integer.parseInt(ageString);

            CalculateMinimumCalories(sex, heightValue, weightValue, ageValue);
        }
    }

    private void CalculateMinimumCalories(String sex, float height, float weight, int age){
        double minimKcal;
        if(sex.equals("Mężczyzna")){
            minimKcal= 66.47+13.7*weight+5.0*height-6.76*age;
        }else{
            minimKcal= 655.1+9.567*weight+1.85*height-4.68*age;
        }

        youCanEat.setText(String.valueOf(Math.round(minimKcal)));
    }


}

File: KOTLIN/app/src/main/java/com/example/bmikotlin/HowManyCalories.kt
package com.example.bmikotlin

import android.os.Bundle
import android.widget.RadioButton
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_how_many_calories.*

class HowManyCalories : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_how_many_calories)

        showWhat.setOnClickListener {
            showMinimalCalories()
        }
    }

    fun showMinimalCalories(){
        val heightString:String = heightHarrisBenedict.text.toString()
        val weightString: String = weightHarrisBenedict.text.toString()
        val ageString: String = ageHarrisBenedict.text.toString()

        if(heightString.isNotEmpty()&&weightString.isNotEmpty()&&ageString.isNotEmpty()){
            val radio: RadioButton= findViewById(radioSexGroup.checkedRadioButtonId)
            calculateMinimumCalories(radio.text.toString() ,heightString.toDouble(),
                                        weightString.toDouble(), ageString.toInt())
        }
        else{
            youCanEatText.text= getString(R.string.enter_values)
        }



    }

    fun calculateMinimumCalories(sex:String, height: Double, weight: Double, age: Int){
        if(sex.equals("mężczyzna")){
           val minimKcal: Double = 66.47 + 13.7 * weight + 5.0 * height - 6.76 * age
            youCanEatText.text= getString(R.string.you_can_eat, String.format("%.0f",minimKcal))
        }else{
            val minimKcal: Double = 655.1 + 9.567 * weight + 1.85 * height - 4.68 * age
            youCanEatText.text=  getString(R.string.you_can_eat, String.format("%.0f",minimKcal))
        }
    }
}
